Rename pos/rot/loc/scl

letheed opened the pull request:
Renames `pos` to `positon`, `rot` to `rotation`, `loc` to `location` and `scl` to `scale`.

See #11119.

letheed committed:
Rename pos to position in user facing methods and variables

Rename user facing methods and variables as well as the corresponding
C++ methods according to the folloming changes:

* pos -> position
* rot -> rotation
* loc -> location

C++ variables are left as is.

hpvb commented:
Good idea, but could you merge these commits into one commit? "Clarify button labels" or something similar?

letheed commented:
Sure I can squash them, I just felt it was cleaner to have them separated. It's all methods, properties and signals though, no button labels so don't see why you'd want that named "button labels".

Well almost. There's 3 button labels that change. BTW I think we should have icons instead of labels for those, since they're not menus.

Noshyaar commented:
Skimming through the diff, your changes are okay. However, it puts files with clang-format errors up for checking so it fails the check. Would be nice to fix those errors too.

letheed commented:
@Noshyaar Yeah, my bad, I didn't run clang-format, I just assumed the files would already be compliant. LMK if there's anything else.

hpvb commented:
I only looked at two of the commits when I suggested the title. If everyone is happy with four commits it's really not a big deal. 

LGTM

letheed commented:
@hpvb Ah! Yes, `scale` is just one `Button` indeed! Thing is, I hadn't discussed nor planned on doing `loc`, `rot` or `scl`, so I wanted it to be front and clear for @Noshyaar to review, and easy for me to drop those commits in case they weren't wanted. I'm totally fine with a squash before merge.

Noshyaar commented on scene/resources/default_theme/default_theme.cpp:
```diff
@@ -883,7 +883,7 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 	theme->set_color("grid_minor", "GraphEdit", Color(1, 1, 1, 0.05));
 	theme->set_color("grid_major", "GraphEdit", Color(1, 1, 1, 0.2));
-	theme->set_constant("bezier_len_pos", "GraphEdit", 80 * scale);
+	theme->set_constant("bezier_len_position", "GraphEdit", 80 * scale);
```
There is `bezier_len_neg`, so I assume `pos` in this case means 'positive'. Not sure if you'd want to refactor theme constants.

Noshyaar commented on scene/resources/curve.cpp:
```diff
@@ -1419,7 +1419,7 @@ void Curve3D::_bind_methods() {
 	/*ADD_PROPERTY( PropertyInfo( Variant::VECTOR3_ARRAY, "points_out"), "set_points_out","get_points_out");
-	ADD_PROPERTY( PropertyInfo( Variant::VECTOR3_ARRAY, "points_pos"), "set_points_pos","get_points_pos");
+	ADD_PROPERTY( PropertyInfo( Variant::VECTOR3_ARRAY, "points_position"), "set_points_position","get_points_position");
 */
```
These lines are commented since the very first commit. I guess it's safe to remove it.

Noshyaar commented on scene/resources/curve.cpp:
```diff
@@ -921,7 +921,7 @@ void Curve2D::_bind_methods() {
 	/*ADD_PROPERTY( PropertyInfo( Variant::VECTOR3_ARRAY, "points_out"), "set_points_out","get_points_out");
-	ADD_PROPERTY( PropertyInfo( Variant::VECTOR3_ARRAY, "points_pos"), "set_points_pos","get_points_pos");
+	ADD_PROPERTY( PropertyInfo( Variant::VECTOR3_ARRAY, "points_position"), "set_points_position","get_points_position");
 */
```
These lines are commented since long time ago. I guess it's safe to remove it.

Noshyaar commented:
I'd suggest renaming the real method names too, to fix some inconsistencies.

https://github.com/letheed/godot/blob/b8ccb621988b1d7da93b72da3eda5d8d412116be/scene/2d/canvas_item.cpp#L999-L1000

```
ClassDB::bind_method(D_METHOD("get_local_mouse_position"), &CanvasItem::get_local_mouse_pos);
ClassDB::bind_method(D_METHOD("get_global_mouse_position"), &CanvasItem::get_global_mouse_position);
```

letheed commented:
I did spend a minute thinking at `bezier_len`, couldn't think of `positive`. -__-

There's a couple other that I wasn't sure about and didn't touch, and would like your input on: [`pos_sec`](https://github.com/godotengine/godot/blob/5f39f3a2dfba500b6c9cd8f87b0dd35949043822/scene/animation/animation_player.cpp#L1241) and [`_track_pos_draw`](https://github.com/godotengine/godot/blob/5f39f3a2dfba500b6c9cd8f87b0dd35949043822/editor/animation_editor.cpp#L3702).

Noshyaar commented:
`pos_sec` maybe position in seconds. I'd change it to just `sec` or `seconds`.
I guess `_track_pos_draw` draws a current time indicator on a track. I'd suggest `_track_bar_draw` and rename the Control name to `track_bar`, or leave it as is.

groud commented:
We discussed the PR in the PR monday.
Could you also rename the corresponding cpp functions when feasible ?

letheed commented:
Yeah, I'll see what I can do.

letheed commented:
Should be good. Renamed only exposed methods. (+1 variable `target_pos` that shadowed a method, I left it in the commit.) For reference:

<details>
 <summary>List of functions still using `pos` but not exposed.</summary>
 <p>

```
cpos
cursor_set_pos
_find_canvas_items_at_pos
_find_control_at_pos
_find_item_at_pos
_find_items_at_pos
_find_next_visible_control_at_pos
get_click_pos
get_constant_pos
get_drop_section_at_pos
get_handle_pos
get_history_pos
_get_mouse_pos
get_name_map_pos
_get_point_from_pos
get_point_view_pos
_get_ray_pos
_get_slot_pos
get_tangent_view_pos
_get_tkpos
get_view_pos
get_word_at_pos
get_world_pos
_gui_find_control_at_pos
_insert_pos
last_pos
mbpos
mouse_to_local_pos
mpos
popup_pos
pos
rpos
seek_pos
set_anim_pos
set_cursor_at_pixel_pos
_set_tkpos
set_window_pos
text_pos
_update_history_pos
video_frames_pos
video_pos
_connpos_update
get_char_pos_for
_get_column_pos_of_word
is_pos_at_end_of_items
make_canvas_pos_local
pos_warped
set_pos_a
set_pos_b
```

</p></details>

akien-mga commented on editor/plugins/canvas_item_editor_plugin.cpp:
```diff
@@ -3422,7 +3422,7 @@ CanvasItemEditor::CanvasItemEditor(EditorNode *p_editor) {
 	animation_hb->hide();
 
-	key_loc_button = memnew(Button("loc"));
+	key_loc_button = memnew(Button("Location"));
```
That and further changes to the CanvasItemEditor will likely mess it up by making the buttons way too wide.
Maybe we can replace those by icons? @djrm

djrm replied:
@letheed submited a the icons, so i guess he can simply put the names here, and wait for the new icons to merge into godot.

letheed replied:
Since we'll have icons I'll just drop the Button changes.

akien-mga commented on core/math/a_star.cpp:
```diff
@@ -425,8 +425,8 @@ void AStar::_bind_methods() {
-	ClassDB::bind_method(D_METHOD("get_closest_point", "to_pos"), &AStar::get_closest_point);
-	ClassDB::bind_method(D_METHOD("get_closest_pos_in_segment", "to_pos"), &AStar::get_closest_pos_in_segment);
+	ClassDB::bind_method(D_METHOD("get_closest_point", "to_position"), &AStar::get_closest_point);
+	ClassDB::bind_method(D_METHOD("get_closest_pos_in_segment", "to_position"), &AStar::get_closest_pos_in_segment);
```
Should be `get_closest_pos*ition*_in_segment` I guess.

letheed replied:
Yeah I noticed that when I did the cpp methods. The rename for this one is in the "method" commit. I left it there for review but I'll do a fixup of both commits before the merge.

akien-mga commented on core/variant_call.cpp:
```diff
@@ -1560,9 +1560,9 @@ void register_variant_methods() {
-	ADDFUNC1(ARRAY, NIL, Array, resize, INT, "pos", varray());
-	ADDFUNC2(ARRAY, NIL, Array, insert, INT, "pos", NIL, "value", varray());
-	ADDFUNC1(ARRAY, NIL, Array, remove, INT, "pos", varray());
+	ADDFUNC1(ARRAY, NIL, Array, resize, INT, "position", varray());
```
I guess that was a buggy name and it should be `size`.

akien-mga commented on editor/animation_editor.cpp:
```diff
@@ -3916,7 +3916,7 @@ AnimationKeyEditor::AnimationKeyEditor() {
 	track_pos->set_mouse_filter(MOUSE_FILTER_IGNORE);
 	track_editor->add_child(track_pos);
-	track_pos->connect("draw", this, "_track_pos_draw");
+	track_pos->connect("track_bar", this, "_track_bar_draw");
```
That's bogus. `"draw"` was the name of the signal to connect.

akien-mga commented on editor/animation_editor.cpp:
```diff
@@ -3700,7 +3700,7 @@ void AnimationKeyEditor::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_clear_selection_for_anim"), &AnimationKeyEditor::_clear_selection_for_anim);
 	ClassDB::bind_method(D_METHOD("_select_at_anim"), &AnimationKeyEditor::_select_at_anim);
-	ClassDB::bind_method(D_METHOD("_track_pos_draw"), &AnimationKeyEditor::_track_pos_draw);
+	ClassDB::bind_method(D_METHOD("_track_bar_draw"), &AnimationKeyEditor::_track_pos_draw);
```
Would be worth changing the name of the C++ method too for clarity.

akien-mga replied:
Or maybe leave the old name, it's not a user facing method, it's just to bind a signal for the editor update. If you *really* want to change it, you should also rename `track_pos` to `track_position` and `timeline_pos` to `timeline_position` etc.

akien-mga commented on scene/2d/canvas_item.cpp:
```diff
@@ -1013,7 +1014,7 @@ void CanvasItem::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("is_transform_notification_enabled"), &CanvasItem::is_transform_notification_enabled);
 
-	ClassDB::bind_method(D_METHOD("make_canvas_pos_local", "screen_point"),
+	ClassDB::bind_method(D_METHOD("make_canvas_position_local", "screen_point"),
```
Please rename the implementation too.

akien-mga replied:
Ah I see it's done in the last commit.

akien-mga commented on scene/2d/line_2d.cpp:
```diff
@@ -269,12 +269,12 @@ void Line2D::_bind_methods() {
 
-	ClassDB::bind_method(D_METHOD("set_point_pos", "i", "pos"), &Line2D::set_point_pos);
-	ClassDB::bind_method(D_METHOD("get_point_pos", "i"), &Line2D::get_point_pos);
+	ClassDB::bind_method(D_METHOD("set_point_position", "i", "position"), &Line2D::set_point_pos);
```
Please rename the implementation for those two.

akien-mga commented on scene/3d/audio_stream_player_3d.cpp:
```diff
@@ -802,12 +802,12 @@ void AudioStreamPlayer3D::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("is_playing"), &AudioStreamPlayer3D::is_playing);
-	ClassDB::bind_method(D_METHOD("get_pos"), &AudioStreamPlayer3D::get_pos);
+	ClassDB::bind_method(D_METHOD("get_position"), &AudioStreamPlayer3D::get_pos);
```
Here too.

akien-mga commented:
Apart from a couple issues, look good. A rebase to fix merge conflicts + squash the commits would be good, then we can review it a last time and merge.

letheed commented:
* Renamed `_track_pos_draw` to `_track_position_draw` in both the C++ method and the binding, for consistency. Didn't touch the variables since I didn't anywhere else.
* Renamed `get_drop_section_at_pos` since it's exposed now.
* Dropped button changes. They'll get icons anyway.
* Re-did the class documentations.
    * Replaced "transform/pos" by "transform/origin" in the Tween.xml example, since I'm guessing those are the member variables. I'd like a confirmation or a correction though.

akien-mga commented:
Can I ask you to rebase a last time to fix the merge conflict, and maybe check that no pos/rot/loc/scl calls were added in the meantime? Then I'll merge it, pinky promise :o)

letheed commented:
Should be good. Aside from conflicts, I only had to touch a handful of class doc files.
